t/plugin/limit-count2.t

```diff
@@ -178,3 +178,137 @@ GET /hello
 --- error_code: 503
 --- response_body
 {"error_msg":"Requests are too frequent, please try again later."}
+
+
+
+=== TEST 6: set key type to var_combination
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "plugins": {
+                        "limit-count": {
+                            "count": 2,
+                            "time_window": 60,
+                            "rejected_code": 503,
+                            "key": "$http_a $http_b",
+                            "key_type": "var_combination"
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: exceed the burst
+--- config
+    location /t {
+        content_by_lua_block {
+            local json = require "t.toolkit.json"
+            local http = require "resty.http"
+            local uri = "http://127.0.0.1:" .. ngx.var.server_port
+                        .. "/hello"
+            local ress = {}
+            for i = 1, 4 do
+                local httpc = http.new()
+                local res, err = httpc:request_uri(uri, {headers = {a = 1}})
+                if not res then
+                    ngx.say(err)
+                    return
+                end
+                table.insert(ress, res.status)
+            end
+            ngx.say(json.encode(ress))
+        }
+    }
+--- request
+GET /t
+--- no_error_log
+[error]
+--- response_body
+[200,200,503,503]
+
+
+
+=== TEST 8: don`t exceed the burst
+--- config
+    location /t {
+        content_by_lua_block {
+            local json = require "t.toolkit.json"
+            local http = require "resty.http"
+            local uri = "http://127.0.0.1:" .. ngx.var.server_port
+                        .. "/hello"
+            local ress = {}
+            for i = 1, 2 do
+                local httpc = http.new()
+                local res, err = httpc:request_uri(uri, {headers = {a = 2}})
```

Comment:
What's the difference between TEST 7/8? Just because TEST 8 iterates 2 steps?

Reply:
What's more, TEST 8 use the other key to count

Reply:
I think if we add two more steps, the result of TEST 8 will be the same as TEST 7?

Reply:
@spacewander Yes it is, so should I remove this case or keep it? Actually I take [this](https://github.com/apache/apisix/pull/5302/files#diff-7645485edca150830424fd3151413a2eddb93d1dcea7fe694ce458113a6b3bf3) as an example.

Reply:
It is:
```
local res, err = httpc:request_uri(uri, {headers = {a = i}})
```
Not `a = 2`. And you should test it three times to trigger the limit exceed.